frontend/src/components/userprofile/userprofile.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import './userprofile.css';
+
+const UserProfile = () => {
+  const [activeTab, setActiveTab] = useState('overview');
+
+  const handleTabClick = (tab) => {
+    setActiveTab(tab);
+  };
+
+  return (
```

Comment:
Looks great just a quick suggestion. Once they go into the profile page, there is no back button or a way for them to switch between the different options. Thats why I created the Navbar. To use it, youll have to import it and then use it... It will look something like this...
```javascript
import Navbar from '../navbar/navbar.jsx'
```
and 
```javascript
<Navbar />
```
I have an example of this inside both the posts route and the home page.